rest/blip_api_test.go

```diff
@@ -3628,3 +3628,113 @@ func TestMinRevPosWorkToAvoidUnnecessaryProveAttachment(t *testing.T) {
+//
+// - Open a one-off subChanges request, ensure it works.
+// - Open a subsequent continuous request, and ensure it works.
+// - Open another continuous subChanges, and asserts that it gets an error on the 2nd one, because the first is still running.
+// - Open another one-off subChanges request, assert we still get an error.
+func TestMultipleOutstandingChangesSubscriptions(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyAll)()
+
+	bt, err := NewBlipTester(t)
+	require.NoError(t, err, "Error creating BlipTester")
+	defer bt.Close()
+
+	bt.blipContext.HandlerForProfile["changes"] = func(request *blip.Message) {
+		if !request.NoReply() {
+			// Send an empty response to avoid the Sync: Invalid response to 'changes' message
+			response := request.Response()
+			emptyResponseVal := []interface{}{}
+			emptyResponseValBytes, err := base.JSONMarshal(emptyResponseVal)
+			assert.NoError(t, err, "Error marshalling response")
+			response.SetBody(emptyResponseValBytes)
+		}
+	}
+
+	pullStats := bt.restTester.GetDatabase().DbStats.CBLReplicationPull()
+	assert.EqualValues(t, 0, pullStats.NumPullReplActiveContinuous.Value())
+	assert.EqualValues(t, 0, pullStats.NumPullReplActiveOneShot.Value())
+
+	// Open an initial continuous = false subChanges request, which we'd expect to release the lock after it's "caught up".
+	subChangesRequest := blip.NewRequest()
+	subChangesRequest.SetProfile("subChanges")
+	subChangesRequest.Properties["continuous"] = "false"
+	subChangesRequest.SetCompressed(false)
+	sent := bt.sender.Send(subChangesRequest)
+	goassert.True(t, sent)
+	subChangesResponse := subChangesRequest.Response()
+	goassert.Equals(t, subChangesResponse.SerialNumber(), subChangesRequest.SerialNumber())
+	errorCode := subChangesResponse.Properties["Error-Code"]
+	log.Printf("errorCode: %v", errorCode)
+	respBody, err := subChangesResponse.Body()
+	require.NoError(t, err)
+	assert.Equal(t, "", errorCode, "resp: %s", respBody)
+
+	pullStats = bt.restTester.GetDatabase().DbStats.CBLReplicationPull()
+	assert.EqualValues(t, 1, pullStats.NumPullReplActiveOneShot.Value())
+
+	// Send continous subChanges to subscribe to changes, which will cause the "changes" profile handler above to be called back
+	subChangesRequest = blip.NewRequest()
+	subChangesRequest.SetProfile("subChanges")
+	subChangesRequest.Properties["continuous"] = "true"
+	subChangesRequest.SetCompressed(false)
+	sent = bt.sender.Send(subChangesRequest)
+	goassert.True(t, sent)
+	subChangesResponse = subChangesRequest.Response()
+	goassert.Equals(t, subChangesResponse.SerialNumber(), subChangesRequest.SerialNumber())
+	errorCode = subChangesResponse.Properties["Error-Code"]
+	log.Printf("errorCode: %v", errorCode)
+	respBody, err = subChangesResponse.Body()
+	require.NoError(t, err)
+	assert.Equal(t, "", errorCode, "resp: %s", respBody)
+
+	pullStats = bt.restTester.GetDatabase().DbStats.CBLReplicationPull()
+	// Might need to expect 0 after CBG-1824
```

Comment:
Not here we don't - I guess this comment was meant to be put up at line 3676 ?

Reply:
Put that there as well as thought if the CBG-1824 fix is to decrement the stat after the changes have been sent, then they will be decremented before the assert gets called.